### DISCUSSION TITLE
Has any way to wait first AsyncData on Stream riverpod?

### DISCUSSION THREAD

**Discussion by normidar** [rrousselGit/riverpod#3307] [Q&A]
I have a Stream riverpod, but I just need the first AsyncData.

When I use `read`, It is always `AsyncLoading`.

When I use `watch`, It will update every time.

Now I had to make Stream as a future to get this feature.

**Comment by normidar** [comment]
> Now I had to make Stream as a future to get this feature.

When I use this solution, I will have two same features riverpod.

**Comment by devbenjoff** [comment]
`await ref.watch(yourStream.future)` maybe?
